.github/workflows/functional-test-noncloud.yaml

```diff
@@ -0,0 +1,445 @@
+            --set de.image=${{ env.DE_IMAGE }},de.tag=${{ env.DE_TAG }} \
+            --set-file global.rootCA.cert=$TEMP_CERT_DIR/certs/${{ env.LOCAL_REGISTRY_SERVER }}/client.crt
+
+          echo "*** Create workspace, group and environment for test ***"
+          rad workspace create kubernetes
+          rad group create kind-radius
+          rad group switch kind-radius
+
+          # The functional test is designed to use default namespace. So you must create the environment for default namespace.
+          rad env create kind-radius --namespace default
+          rad env switch kind-radius
+
+      - name: Publish Terraform test recipes
+        run: |
+          make publish-test-terraform-recipes
+
+      - name: Publish Bicep Test Recipes
+        run: |
+          export PATH=$GITHUB_WORKSPACE/bin:$PATH
+          which rad || { echo "cannot find rad"; exit 1; }
+          make publish-test-bicep-recipes
+        env:
+          BICEP_RECIPE_REGISTRY: "${{ env.LOCAL_REGISTRY_SERVER }}:${{ env.LOCAL_REGISTRY_PORT }}"
+          BICEP_RECIPE_TAG_VERSION: ${{ env.REL_VERSION }}
+          SSL_CERT_FILE: $TEMP_CERT_DIR/certs/${{ env.LOCAL_REGISTRY_SERVER }}/client.crt
+
+      - name: Run functional tests
+        run: |
+          # Ensure rad cli is in path before running tests.
+          export PATH=$GITHUB_WORKSPACE/bin:$PATH
+          # Make directory to capture functional test results
+          mkdir -p ./dist/functional_test 
+          cd $GITHUB_WORKSPACE
+
+          which rad || { echo "cannot find rad"; exit 1; }
+
+          make test-functional-${{ matrix.name }}
+        env:
+          DOCKER_REGISTRY: "${{ env.LOCAL_REGISTRY_NAME }}:${{ env.LOCAL_REGISTRY_PORT }}"
+          TEST_TIMEOUT: ${{ env.FUNCTIONALTEST_TIMEOUT }}
+          RADIUS_CONTAINER_LOG_PATH: ${{ github.workspace }}/${{ env.RADIUS_CONTAINER_LOG_BASE }}
+          RADIUS_SAMPLES_REPO_ROOT: ${{ github.workspace }}/samples
+          BICEP_RECIPE_REGISTRY: "${{ env.LOCAL_REGISTRY_NAME }}:${{ env.LOCAL_REGISTRY_PORT }}"
+          BICEP_RECIPE_TAG_VERSION: ${{ env.BICEP_RECIPE_TAG_VERSION }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GOTESTSUM_OPTS: "--junitfile ./dist/functional_test/results.xml"
+
+      - name: Process Functional Test Results
+        uses: ./.github/actions/process-test-results
+        # In case of failure, upload functional_test_results to artifacts so that they are not erased by subsequent runs.
+        if: failure() && github.repository == 'radius-project/radius'
+        with:
+          test_group_name: "Functional Tests - ${{ matrix.name }}"
+          artifact_name: "functional_test_results_${{ matrix.name }}"
+          result_directory: "dist/functional_test/"
+
+      - uses: azure/setup-kubectl@v4
+        if: always()
+        with:
+          version: ${{ env.KUBECTL_VER }}
```

Comment:
Do we need this here? I think the scripts already called into kubectl. IMO it's better to install dependencies early in the workflow instead of near the end.

Reply:
I removed this step from both workflows.